spring-cloud-aws-sns/src/main/java/io/awspring/cloud/sns/sms/core/DefaultMessageAttributeConverter.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2013-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.awspring.cloud.sns.sms.core;
+
+import static io.awspring.cloud.sns.core.MessageAttributeDataTypes.NUMBER;
+import static io.awspring.cloud.sns.core.MessageAttributeDataTypes.STRING;
+
+import io.awspring.cloud.sns.sms.attributes.*;
+import java.util.HashMap;
+import java.util.Map;
+import software.amazon.awssdk.services.sns.model.MessageAttributeValue;
+
+/**
+ * @author Matej Nedic
+ * @since 3.0.0
+ */
+public class DefaultMessageAttributeConverter implements MessageAttributeConverter {
+
+	public Map<String, MessageAttributeValue> convert(SmsMessageAttributes attr) {
```

Comment:
Have you considered moving this conversion to `SmsMessageAttributes` class? Does not seem like there can be any other reasonable implementation of `MessageAttributeConverter` interface